src/components/reusable/CityItem/CityItem.jsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import Grid from '@material-ui/core/Grid';
+import WeatherIcon from '../../reusable/WeatherIcon/WeatherIcon';
+import { Link } from "react-router-dom";
+import {
+    cityItemWrapper,
+    windWrapper,
+    gridItem,
+    windText,
+    city,
+    temperature
+} from './CityItem.scss';
+import { setCurrentCity } from '../../../actions/cityActions';
+
+class CityItem extends React.Component {
+    cityData = null;
+
+    constructor(props) {
+        super(props);
+        this.cityData = this.props.cityData;
+        this.handleClick = this.handleClick.bind(this);
+    }
+
+    handleClick() {
+        this.cityData.isCurrentCity = true;
```

Comment:
be carefull with mutations... they could become confusing not only for you...

You should just track it via redux. You are already dispatching an action to setCurrent city. Just change respective flag via reducer.